archinstall/lib/disk.py

```diff
@@ -14,10 +13,166 @@
 MBR = 0b00000010
 
 
-# import ctypes
-# import ctypes.util
-# libc = ctypes.CDLL(ctypes.util.find_library('c'), use_errno=True)
-# libc.mount.argtypes = (ctypes.c_char_p, ctypes.c_char_p, ctypes.c_char_p, ctypes.c_ulong, ctypes.c_char_p)
+def valid_fs_type(fstype :str) -> bool:
+	# https://www.gnu.org/software/parted/manual/html_node/mkpart.html
+
+	return fstype in [
+		"ext2",
```

Comment:
Why just ext2 and not ext3 or ext4 here?

Reply:
This should work: 

sudo parted -a opt /dev/sda mkpart primary ext4 0% 100%

Reply:
`ext4` gets ignored I presume as it's not a [supported filesystem](https://www.gnu.org/software/parted/manual/html_node/mkpart.html).